### PR TITLE
feat: set up RPC

### LINKED ISSUE #190: Set up Hono's RPC
## Description

To seamlessly connect the backend with the frontend, we'll utilize Hono's RPC.

## Criteria

- [ ] Set up RPC
- [ ] Test to see if it's working correctly

## Notes

https://hono.dev/docs/guides/rpc

### PR DESCRIPTION
**Pull request by akiomatic**
## Overview

I’ve successfully added Hono’s RPC, allowing for easy sharing of API specifications between the backend and frontend!

## Changes

- Added Hono's RPC
- Refactored `/api` entirely to follow [Hono's guide](https://hono.dev/docs/guides/rpc#using-rpc-with-larger-applications)
- Added a new environment variable called `NEXT_PUBLIC_APPLICATION_URL`

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)
- [x] The branch name follows our branch naming rules
- [x] The PR title follows our PR title rules
- [x] My code follows our coding style
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [ ] Code readability and simplicity
- [ ] Follows best practices and coding standards
- [ ] Understandable and maintainable code

### CODE REVIEW

File: src/app/api/[[...route]]/route.ts [hunk 1]
```diff
@@ -10,45 +10,48 @@ import categoryRoute from './routes/category.route';
...
+    }
+
+    const landlord = await prisma.landlord.findUnique({
+      where: { id: session.user.id },
+    });
 
-  if (!session) {
     return c.json({
-      message: 'You are not logged in!',
+      message: 'You are logged in!',
```
Comment (KaiKoide) [inline src/app/api/[[...route]]/route.ts:45]: Could you change the `message` to the `error` for the type safety?

**Review by KaiKoide — CHANGES_REQUESTED** [review]
Thank you for your work! please check my comment.

File: src/app/api/[[...route]]/route.ts (hunk 1 above)
Comment (akiomatic) [reply src/app/api/[[...route]]/route.ts:45]: Hi. Thank you. I've fixed it.

**Review by KaiKoide — APPROVED** [review]
LGTM!

**Pull request merged**